src/app/applications/components/application-sync-panel/application-sync-panel.tsx

```diff
@@ -15,9 +15,8 @@ export const ApplicationSyncPanel = ({application, selectedResource, hide}: {
 
     const [form, setForm] = React.useState<FormApi>(null);
     const isVisible = !!(selectedResource && application);
-    const syncResIndex = application && selectedResource && application.status.resources.findIndex((item) => {
-        return nodeKey(item) === selectedResource;
-    });
+    const appResources = (application && selectedResource && application.status.resources || []).sort((first, second) => nodeKey(first).localeCompare(nodeKey(second)));
```

Comment:
I think you also want `&& application.status`

Reply:
you are right. Thank you!